src/services/refactors/extractType.ts

```diff
@@ -229,6 +229,8 @@ namespace ts.refactor {
     function doTypedefChange(changes: textChanges.ChangeTracker, file: SourceFile, name: string, info: ExtractInfo) {
         const { firstStatement, selection, typeParameters } = info;
 
+        removeCommentsFromTypeNode(selection);

```

Comment:
Would setting `EmitFlags.NoComments | EmitFlags.NoNestedComments` work instead?

Reply:
that is definitely better 👍